src/akkudoktoreos/devices/battery.py

```diff
@@ -1,40 +1,45 @@
-from typing import Optional
+from typing import Any, Optional
 
 import numpy as np
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, Field, field_validator
 
+from akkudoktoreos.utils.utils import NumpyEncoder
 
-def max_ladeleistung_w_field(default=None):
+
+def max_ladeleistung_w_field(default: Optional[float] = None) -> Optional[float]:
     return Field(
-        default,
+        default=default,
         gt=0,
         description="An integer representing the charging power of the battery in watts.",
     )
```

Comment:
I'm not familiar with pydantic, but shouldn't the return type hint be `Field`?

Reply:
Field is actually a function but it returns a FieldInfo. The type annotation for the Field function however is Any as return type as it is assigned to the pydantic model field (with a user type like Optional[float]). Therefore this type hint cheats a bit in the way similar as it's done with the deap.creator.Individual -> list[int].